### PR TITLE
MM-60732 - Scheduled posts channel link now takes you to the first scheduled post in channel/thread in list

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Scheduled posts channel link now takes you to the first scheduled post in channel/thread in list

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-60732

#### Release Note
```release-note
none
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]
LGTM, just two small potential improvements.

File: webapp/channels/src/components/drafts/draft_row.tsx [hunk 1]
```diff
@@ -245,6 +248,8 @@ function DraftRow({
         }
     }, [thread?.id]);
 
+    const alertRef = useScrollOnRender();
```
[the remark this answers is not in the document]
Comment (harshilsharma63) [reply webapp/channels/src/components/drafts/draft_row.tsx:251]: So so I either create refs for all scheduled post list items and then choose the ref to scroll into view, or I only create and apply the ref for the one item I wanna scroll into view. I didn't see much benefit for the first approach so went with the this approach. Yeah it doesn't look nice in the first time, but makes sense.

File: webapp/channels/src/components/common/hooks/scroll.ts
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/common/hooks/scroll.ts:1]: nit : use_scroll_on_render.ts

File: webapp/channels/src/components/common/hooks/scroll.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+export function useScrollOnRender() {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/common/hooks/scroll.ts:6]: can you include a comment on what this function is and whats its purpose

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
first scheduled message gets highlighted when we click on See all

File: webapp/channels/src/components/common/hooks/scroll.ts
Comment (harshilsharma63) [reply webapp/channels/src/components/common/hooks/scroll.ts:1]: Done.

File: webapp/channels/src/components/common/hooks/scroll.ts (hunk 2 above)
Comment (harshilsharma63) [reply webapp/channels/src/components/common/hooks/scroll.ts]: Done.

**Review by M-ZubairAhmed — APPROVED** [review]

File: webapp/channels/src/components/drafts/panel/panel.tsx [hunk 3]
```diff
@@ -12,6 +12,8 @@ type Props = {
     children: ({hover}: {hover: boolean}) => React.ReactNode;
     onClick: () => void;
     hasError: boolean;
+    innerRef?: React.Ref<HTMLElement>;
+    className?: string;
```
[the remark this answers is not in the document]
Comment (harshilsharma63) [reply webapp/channels/src/components/drafts/panel/panel.tsx]: Done.

**Pull request merged**